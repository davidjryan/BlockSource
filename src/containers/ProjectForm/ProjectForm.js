import React, { Component } from 'react';

import RaisedButton from 'material-ui/RaisedButton';

import './ProjectForm.css'

export default class projectForm extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      description: '',
      goal: ''
    }
  }

  handleChange = (event) => {
    const target = event.target;
  }

  render() {
    return(
      <div className='project-form-container'>
        <div className='project-form'>
          <div className='input-labels'>
            <span className='project-input-label'>Title</span>
            <span className='project-input-label'>Description</span>
            <span className='project-input-label'>Goal</span>
          </div>
          <div className='project-inputs'>
            <input 
              name='title'
              type='text' 
              className='project-input single-input'
            />
            <textarea
              name='description'
              type='text' 
              className='project-input'
              placeholder='Why should people fund this project?'
            >
            </textarea>
            <input
              name='goal'
              type='text' 
              className='project-input single-input'
            />
        </div>
      </div>
        <RaisedButton label='default'/>
        <button className='button btn-create'>Create</button>
    </div>
    )
  }

}

