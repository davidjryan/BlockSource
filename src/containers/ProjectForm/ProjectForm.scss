@import '../../style/variable.scss';

.project-form-container {
  display: flex;
  flex-direction: column;
  align-items: center;

}
.project-form {
  display: flex;
  justify-content: center;
}

.input-labels {
  display: flex;
  flex-direction: column;
  text-align: right;
  margin-right: 25px;
}

.project-inputs {
  display: flex;
  flex-direction: column;
}
.project-input {
  width: 300px;
  margin: 5px;
  font-size: 16px;
}
.project-input-label {
  margin: 10px;
} 

.single-input {
  height: 20px;
}

button.btn-create {
  border: 1px solid green;
  width: 150px;
  margin-top: 20px;
}