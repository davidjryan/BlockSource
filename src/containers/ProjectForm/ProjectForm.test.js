import React from 'react';
import { shallow } from 'enzyme';
import ProjectForm from './ProjectForm';

describe('Projects tests', () => {
  let projectForm;

  beforeEach( () => {
    projectForm = shallow(<ProjectForm />)
  })

  it('should match snapshot', () => {
    expect(projectForm).toMatchSnapshot()
  })
})