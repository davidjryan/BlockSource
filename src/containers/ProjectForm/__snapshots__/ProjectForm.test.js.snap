// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <projectForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="project-form"
>
          <div
                    className="input-labels"
          >
                    <span
                              className="project-input-label"
                    >
                              Title
                    </span>
                    <span
                              className="project-input-label"
                    >
                              Description
                    </span>
                    <span
                              className="project-input-label"
                    >
                              Goal
                    </span>
          </div>
          <div
                    className="project-inputs"
          >
                    <input
                              className="project-input single-input"
                              name="title"
                              type="text"
                    />
                    <textarea
                              className="project-input"
                              name="description"
                              placeholder="Why should people fund this project?"
                              type="text"
                    />
                    <input
                              className="project-input single-input"
                              name="goal"
                              type="text"
                    />
          </div>
</div>,
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="default"
          labelPosition="after"
          primary={false}
          secondary={false}
/>,
        <button
          className="button btn-create"
>
          Create
</button>,
      ],
      "className": "project-form-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="input-labels"
>
              <span
                            className="project-input-label"
              >
                            Title
              </span>
              <span
                            className="project-input-label"
              >
                            Description
              </span>
              <span
                            className="project-input-label"
              >
                            Goal
              </span>
</div>,
            <div
              className="project-inputs"
>
              <input
                            className="project-input single-input"
                            name="title"
                            type="text"
              />
              <textarea
                            className="project-input"
                            name="description"
                            placeholder="Why should people fund this project?"
                            type="text"
              />
              <input
                            className="project-input single-input"
                            name="goal"
                            type="text"
              />
</div>,
          ],
          "className": "project-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="project-input-label"
>
                  Title
</span>,
                <span
                  className="project-input-label"
>
                  Description
</span>,
                <span
                  className="project-input-label"
>
                  Goal
</span>,
              ],
              "className": "input-labels",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Title",
                  "className": "project-input-label",
                },
                "ref": null,
                "rendered": "Title",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description",
                  "className": "project-input-label",
                },
                "ref": null,
                "rendered": "Description",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Goal",
                  "className": "project-input-label",
                },
                "ref": null,
                "rendered": "Goal",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="project-input single-input"
                  name="title"
                  type="text"
/>,
                <textarea
                  className="project-input"
                  name="description"
                  placeholder="Why should people fund this project?"
                  type="text"
/>,
                <input
                  className="project-input single-input"
                  name="goal"
                  type="text"
/>,
              ],
              "className": "project-inputs",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "project-input single-input",
                  "name": "title",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "project-input",
                  "name": "description",
                  "placeholder": "Why should people fund this project?",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "project-input single-input",
                  "name": "goal",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "fullWidth": false,
          "label": "default",
          "labelPosition": "after",
          "primary": false,
          "secondary": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Create",
          "className": "button btn-create",
        },
        "ref": null,
        "rendered": "Create",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="project-form"
>
            <div
                        className="input-labels"
            >
                        <span
                                    className="project-input-label"
                        >
                                    Title
                        </span>
                        <span
                                    className="project-input-label"
                        >
                                    Description
                        </span>
                        <span
                                    className="project-input-label"
                        >
                                    Goal
                        </span>
            </div>
            <div
                        className="project-inputs"
            >
                        <input
                                    className="project-input single-input"
                                    name="title"
                                    type="text"
                        />
                        <textarea
                                    className="project-input"
                                    name="description"
                                    placeholder="Why should people fund this project?"
                                    type="text"
                        />
                        <input
                                    className="project-input single-input"
                                    name="goal"
                                    type="text"
                        />
            </div>
</div>,
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="default"
            labelPosition="after"
            primary={false}
            secondary={false}
/>,
          <button
            className="button btn-create"
>
            Create
</button>,
        ],
        "className": "project-form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="input-labels"
>
                <span
                                className="project-input-label"
                >
                                Title
                </span>
                <span
                                className="project-input-label"
                >
                                Description
                </span>
                <span
                                className="project-input-label"
                >
                                Goal
                </span>
</div>,
              <div
                className="project-inputs"
>
                <input
                                className="project-input single-input"
                                name="title"
                                type="text"
                />
                <textarea
                                className="project-input"
                                name="description"
                                placeholder="Why should people fund this project?"
                                type="text"
                />
                <input
                                className="project-input single-input"
                                name="goal"
                                type="text"
                />
</div>,
            ],
            "className": "project-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="project-input-label"
>
                    Title
</span>,
                  <span
                    className="project-input-label"
>
                    Description
</span>,
                  <span
                    className="project-input-label"
>
                    Goal
</span>,
                ],
                "className": "input-labels",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Title",
                    "className": "project-input-label",
                  },
                  "ref": null,
                  "rendered": "Title",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description",
                    "className": "project-input-label",
                  },
                  "ref": null,
                  "rendered": "Description",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Goal",
                    "className": "project-input-label",
                  },
                  "ref": null,
                  "rendered": "Goal",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="project-input single-input"
                    name="title"
                    type="text"
/>,
                  <textarea
                    className="project-input"
                    name="description"
                    placeholder="Why should people fund this project?"
                    type="text"
/>,
                  <input
                    className="project-input single-input"
                    name="goal"
                    type="text"
/>,
                ],
                "className": "project-inputs",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "project-input single-input",
                    "name": "title",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "project-input",
                    "name": "description",
                    "placeholder": "Why should people fund this project?",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "project-input single-input",
                    "name": "goal",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "fullWidth": false,
            "label": "default",
            "labelPosition": "after",
            "primary": false,
            "secondary": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create",
            "className": "button btn-create",
          },
          "ref": null,
          "rendered": "Create",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
